### PR TITLE
Feature/asv 217 links to privacy and terms conditions

### PR DESCRIPTION
**Pull request by paquiquinho**
**What does this PR do?**

This PR aims to add links to terms and conditions and to privacy policy to comply with the new GDPR rules;

**Database changed?**

No

**Where should the reviewer start?**

- [ ] LoginSignUpCredentialsFragment.java
- [ ] SettingsFragment.java

**How should this be manually tested?**

These links are displayed bellow the join button when signing up and at the bottom of the settings menu. Find them and click them. You should be taken to a webview within the app displaying either terms and conditions or privacy policy

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-217](<https://aptoide.atlassian.net/browse/ASV-217>)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Partners build
- [ ] Unit tests pass
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java [hunk 1]
```diff
@@ -320,9 +334,38 @@ private void setAptoideSignUpLoginAreaVisible() {
...
-    termsAndConditions = (TextView) view.findViewById(R.id.terms_and_conditions);
     separator = view.findViewById(R.id.separator);
 
+    ClickableSpan clickableSpan = new ClickableSpan() {
+      @Override public void onClick(View view) {
+        termsAndConditionsClicked.onNext(null);
+      }
+    };
+
+    ClickableSpan clickableSpan2 = new ClickableSpan() {
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java]: Naming

File: app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java [hunk 2]
```diff
@@ -320,9 +334,38 @@ private void setAptoideSignUpLoginAreaVisible() {
     }
     loginArea = view.findViewById(R.id.login_button_area);
     signUpArea = view.findViewById(R.id.sign_up_button_area);
-    termsAndConditions = (TextView) view.findViewById(R.id.terms_and_conditions);
     separator = view.findViewById(R.id.separator);
 
+    ClickableSpan clickableSpan = new ClickableSpan() {
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java]: Naming

File: app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java [hunk 3]
```diff
@@ -320,9 +334,38 @@ private void setAptoideSignUpLoginAreaVisible() {
...
+    ClickableSpan clickableSpan2 = new ClickableSpan() {
+      @Override public void onClick(View view) {
+        privacyPolicyClicked.onNext(null);
+      }
+    };
+
+    String baseString = getString(R.string.terms_and_conditions_privacy_sign_up_message);
+    String termsAndConditionsPlaceHolder = getString(R.string.settings_terms_conditions);
+    String privacyPolicyPlaceHolder = getString(R.string.settings_privacy_policy);
+    String compositeString =
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java]: Naming for compositeString, maybe privacyAndTermsPlaceHolder

File: app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java [hunk 4]
```diff
@@ -320,9 +334,38 @@ private void setAptoideSignUpLoginAreaVisible() {
...
+        String.format(baseString, termsAndConditionsPlaceHolder, privacyPolicyPlaceHolder);
+
+    SpannableString compositeSpan = new SpannableString(compositeString);
+    compositeSpan.setSpan(clickableSpan, compositeString.indexOf(termsAndConditionsPlaceHolder),
+        compositeString.indexOf(termsAndConditionsPlaceHolder)
+            + termsAndConditionsPlaceHolder.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+    compositeSpan.setSpan(clickableSpan2, compositeString.indexOf(privacyPolicyPlaceHolder),
+        compositeString.indexOf(privacyPolicyPlaceHolder) + privacyPolicyPlaceHolder.length(),
+        Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java]: Maybe put this in a method that receives the String as an argument
Naming compositeString, maybe privacyAndTermsPlaceholderSpan

File: app/src/main/java/cm/aptoide/pt/presenter/LoginSignUpCredentialsPresenter.java [hunk 5]
```diff
@@ -71,6 +72,26 @@ public LoginSignUpCredentialsPresenter(LoginSignUpCredentialsView view,
     handleTogglePasswordVisibility();
   }
 
+  private void handleClickOnTermsAndConditions() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.termsAndConditionsClickEvent())
+        .doOnNext(__ -> accountNavigator.navigateToTermsAndConditions())
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/presenter/LoginSignUpCredentialsPresenter.java]: After the flatMap of the click it should have the observeOn on MainThread and retry()

File: app/src/main/java/cm/aptoide/pt/presenter/LoginSignUpCredentialsPresenter.java [hunk 6]
```diff
@@ -71,6 +72,26 @@ public LoginSignUpCredentialsPresenter(LoginSignUpCredentialsView view,
...
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnPrivacyPolicy() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.privacyPolicyClickEvent())
+        .doOnNext(__ -> accountNavigator.navigateToPrivacyPolicy())
```
Comment (JDSM01) [inline app/src/main/java/cm/aptoide/pt/presenter/LoginSignUpCredentialsPresenter.java]: After the flatMap of the click it should have the observeOn on MainThread and retry()

File: app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java (hunk 1 above)
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java]: fixed

File: app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java (hunk 2 above)
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/account/view/LoginSignUpCredentialsFragment.java]: fixed

File: app/src/main/java/cm/aptoide/pt/presenter/LoginSignUpCredentialsPresenter.java (hunk 5 above)
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/presenter/LoginSignUpCredentialsPresenter.java]: fixed

File: app/src/main/java/cm/aptoide/pt/presenter/LoginSignUpCredentialsPresenter.java (hunk 6 above)
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/presenter/LoginSignUpCredentialsPresenter.java]: fixed

**Pull request merged**